File: Tugas-Modul-ROS/src/publisher.cpp
#include "ros/ros.h"
#include "demo_package/suratcinta.h"

std::string presensi;
int main(int argc, char ** argv){
    ros::init(argc,argv,"publisher_node");

    ros::NodeHandle nh;

    ros::Publisher data_publisher = nh.advertise<demo_package::suratcinta>("topik_utama_Cplusplus",1000);

    ros::Rate loop_rate(1);

    while(ros::ok()){

        demo_package::suratcinta njir;

        njir.usia = 18;
        njir.tinggi = 1.65;
        njir.nama = "Yanto Sigalo Galo",
        njir.kehadiran = false;
        if(njir.kehadiran == true){
            presensi = "hadir";
        }
        else{
            presensi = "tidak hadir";
        }
       
        ROS_INFO("Usia : %ld", njir.usia);
        ROS_INFO("Tinggi : %lf", njir.tinggi);
        ROS_INFO("Nama : %s", njir.nama.c_str());
        ROS_INFO("Kehadiran : %s", presensi.c_str());

        data_publisher.publish(njir);

        ros::spinOnce();
        loop_rate.sleep();

    }
    return 0;
}

File: Tugas-Modul-ROS/script/publisher.py
import rospy
from demo_package.msg import suratcinta

def talker():
    message_pub = rospy.Publisher("topik_utama_python", suratcinta, queue_size=10)

    rospy.init_node("publisherpy_node", anonymous = True)

    rate = rospy.Rate(1)

    while not rospy.is_shutdown():

        bjir = suratcinta()

        bjir.usia = 21
        bjir.tinggi = 3.14
        bjir.nama = "Pi Nasution"
        bjir.kehadiran = True
        if bjir.kehadiran == True:
            presensi = "hadir"
        else:
            presensi = "tidak hadir"

        rospy.loginfo("Usia : %ld" % bjir.usia)
        rospy.loginfo("Tinggi : %lf" % bjir.tinggi)
        rospy.loginfo("Nama : %s" % bjir.nama)
        rospy.loginfo("Kehadiran : %s" % presensi)

        message_pub.publish(bjir)

        rate.sleep()

if __name__ == "__main__":
    try:
        talker()
    except rospy.ROSInterruptException:
        pass